tests/src/test/scala/cats/collections/PairingHeapSuite.scala

```diff
@@ -0,0 +1,226 @@
+    }
+
+  property("fromIterable is the same as adding") =
+    forAll { (init: List[Int]) =>
+      val heap1 = PairingHeap.fromIterable(init)
+      val heap2 = init.foldLeft(PairingHeap.empty[Int])(_.add(_))
+
+      heap1.toList == heap2.toList
+    }
+
+  property("minimumOption after removing one is >= before") =
+    forAll { (heap: PairingHeap[Int]) =>
+      val min0 = heap.minimumOption
+      val min1 = heap.remove.minimumOption
+
+      (min0, min1) match {
+        case (None, next) => next.isEmpty
+        case (_, None) => heap.size == 1
+        case (Some(m0), Some(m1)) =>
+          m0 <= m1
+      }
+    }
+
+  property("PairingHeap.minimumOption is the real minimum") = {
+    def heapLaw(heap: PairingHeap[Int]): Boolean =
+      heap.minimumOption match {
+        case None => heap.isEmpty
+        case Some(min) =>
+          val heap1 = heap.remove
+          heap1.isEmpty || {
+            min <= heap1.toList.min
+          }
+      }
+
+    forAll { (heap: PairingHeap[Int]) =>
+      val a = heapLaw(heap)
+      // even after removing this is true
+      val b = heapLaw(heap.remove)
+      val c = heapLaw(heap.remove.remove)
+      val d = heapLaw(heap.remove.remove.remove)
+
+      a && b && c && d && PairingHeap.empty[Int].minimumOption.isEmpty
+    }
+  }
+
+  property("PairingHeap.foldLeft is consistent with toList.foldLeft") =
+    forAll { (heap: PairingHeap[Int], init: Long, fn: (Long, Int) => Long) =>
+      heap.foldLeft(init)(fn) == heap.toList.foldLeft(init)(fn)
+    }
+
+  property("Show[PairingHeap[Int]] works like toList.mkString") =
+    forAll { (heap: PairingHeap[Int]) =>
+      Show[PairingHeap[Int]].show(heap) == heap.toList.mkString("PairingHeap(", ", ", ")")
+    }
+
+  property("Order[PairingHeap[Int]] works like List[Int]") =
+    forAll { (a: PairingHeap[Int], b: PairingHeap[Int]) =>
+      Order[PairingHeap[Int]].compare(a, b) == Order[List[Int]].compare(a.toList, b.toList)
+    }
+
+  property("PairingHeap.exists is correct") =
+    forAll { (a: PairingHeap[Int], fn: Int => Boolean) =>
+      a.exists(fn) == a.toList.exists(fn)
+    }
+
+  property("PairingHeap.forall is correct") =
+    forAll { (a: PairingHeap[Int], fn: Int => Boolean) =>
+      a.forall(fn) == a.toList.forall(fn)
+    }
+
+  property("PairingHeap.empty is less than nonEmpty") =
+    forAll { (item: Int, heap: PairingHeap[Int]) =>
+      val ord = Order[PairingHeap[Int]]
+      val a = ord.lteqv(PairingHeap.empty, heap)
+      val b = ord.lt(PairingHeap.empty, PairingHeap.empty + item)
+      val c = ord.gteqv(heap, PairingHeap.empty)
+      val d = ord.gt(PairingHeap.empty + item, PairingHeap.empty)
+
+      a && b && c && d
+    }
+
+  def genPairingHeapLimitedSize(limit: Int): Gen[PairingHeap[Int]] =
+    Gen.choose(0, limit).flatMap(heapGen[Int](_, Arbitrary.arbitrary[Int]))
+
+  property("PairingHeap.combineAll (remove) does < log_2 N + 8 work") =
+    forAll(genPairingHeapLimitedSize(10)) { heap: PairingHeap[Int] =>
```

Comment:
In the previous version based on `Scalatest`, the heap have generated with max size 10 (although there is no explicit limit in `Gen[A]` I saw it when the manually printing size of the heap). So, I've added the explicit limit on heap size because it often was a length bigger than 10 in the new version based on `MUnit`.